linter/src/rules/constraint_missing_not_valid.rs

```diff
@@ -9,9 +9,9 @@ use squawk_parser::ast::{
 
 /// Return list of spans for offending transactions. From the start of BEGIN to
 /// the end of COMMIT.
-fn not_valid_validate_in_transaction(tree: &[RawStmt]) -> Vec<Span> {
+fn not_valid_validate_in_transaction(tree: &[RawStmt], assume_in_transaction: bool) -> Vec<Span> {
     let mut not_valid_names = HashSet::new();
-    let mut in_transaction = false;
```

Comment:
also curious what you think about how we should handle nested transactions here as well

Reply:
like:

```
squawk --assume-in-transaction foo.sql
```

where foo.sql is

```sql
select 1;
begin;
select 2;
commit;
```

Reply:
I reviewed this again and found an edge case that I previously didn't recognize that result in the violation being flagged twice when both --assume-in-transaction was used and an explicit COMMIT was present. I fixed that and wrote a new test for that in 0887e80.

Then I thought about it a little more and changed how that error was being attributed in a396271 (see the commit message for details). This passes tests and should work in the same way, but attribute the violation to the VALIDATE CONSTRAINT statement.